apps/io-wallet-user-func/src/infra/pid-issuer/client.ts

```diff
@@ -0,0 +1,84 @@
+import { PidIssuerApiClientConfig } from "@/app/config";
+import { CredentialRepository } from "@/credential";
+import { FiscalCode } from "@pagopa/ts-commons/lib/strings";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { Agent, RequestInit, fetch } from "undici";
+
+import { PidIssuerHealthCheck } from "./health-check";
+
+export class PidIssuerClient
+  implements CredentialRepository, PidIssuerHealthCheck
+{
+  #baseURL: string;
+  #init: RequestInit;
+  #walletProviderEntity: string;
+
+  healthCheck = () =>
+    TE.tryCatch(
+      async () => {
+        const result = await fetch(new URL(`/revokeAll`, this.#baseURL), {
+          body: JSON.stringify({
+            unique_id: "foo",
+            wallet_provider: this.#walletProviderEntity,
+          }),
+          method: "POST",
+          signal: AbortSignal.timeout(3000),
+          ...this.#init,
+        });
+        return result.status === 404;
+      },
+      (error) => new Error(`error checking PID issuer health: ${error}`),
+    );
+
+  revokeAllCredentials = (fiscalCode: FiscalCode) =>
+    pipe(
+      TE.tryCatch(
+        async () => {
+          const result = await fetch(new URL("/revokeAll", this.#baseURL), {
+            body: JSON.stringify({
+              unique_id: `TINIT-${fiscalCode}`,
+              wallet_provider: this.#walletProviderEntity,
+            }),
+            method: "POST",
+            signal: AbortSignal.timeout(3000),
+            ...this.#init,
+          });
+
+          if (result.status === 404) {
+            // if the credentials have already been revoked the status is 404 but our endpoint is idempotent
+            // add log
+            return undefined;
+          }
```

Comment:
Add a tood. Why not return an error anyway?

Reply:
the IPZS endpoint returns 404 if:

the credentials have already been revoked => I hide the error because our endpoint is idempotent
the credentials do not exist => I hide the error because our endpoint returns 204 if you try to revoke a non-existent one

but we can think about changing this behavior if we want

Reply:
Ok thanks. Maybe add this in the comment

Reply:
right!